classes/controller/AdminController.php

```diff
@@ -4882,4 +4882,19 @@ private function prepareNumberSpecifications(Context $context)
             $numberSpecification->toArray()
         );
     }
+
+    /**
+     * Using this method will display the given message as an "alert info" message at the top of the current
+     * administration page.
+     *
+     * @see admin-dev/themes/default/template/content.tpl
+     *
+     * @param string $message
+     *
+     * @return void
+     */
+    protected function setTopAlertInfoMessage($message)
+    {
+        $this->context->smarty->assign('alertInfo', $message);
```

Comment:
Isn't too overkill?
Should it be more generic?
```php
protected function addTopAlertMessage($message, $namespace = 'info') // Maybe use const here
{
    $messages = $this->context->smarty->getVariable('alertMessages')->value;
    if ($messages === null) {
        $messages = [];
    }

    if (!isset($messages[$namespace])) {
        $messages[$namespace] = [];
    }

    $messages[$namespace][] = $messages;

    $this->context->smarty->assign('alertMessages', $messages);
}
```

Reply:
I don't think its overkill, the place where I was asked to put this info message text seems pretty generic to all legacy admin templates...  So it seems wrong to implement something specific coming only from AdminSuppliersController at this place.

But your suggestion is indeed more generic, I didn't think about all the kinds of info messages, I'll implement this !

Reply:
done !